src/analysis/normed/group/seminorm.lean

```diff
@@ -44,19 +48,31 @@ variables {ι R R' E F G : Type*}
 
 /-- A seminorm on an additive group `G` is a function `f : G → ℝ` that preserves zero, is
 subadditive and such that `f (-x) = f x` for all `x`. -/
+@[protect_proj]
 structure add_group_seminorm (G : Type*) [add_group G] extends zero_hom G ℝ :=
 (add_le' : ∀ r s, to_fun (r + s) ≤ to_fun r + to_fun s)
 (neg' : ∀ r, to_fun (-r) = to_fun r)
 
 /-- A seminorm on a group `G` is a function `f : G → ℝ` that sends one to zero, is submultiplicative
 and such that `f x⁻¹ = f x` for all `x`. -/
-@[to_additive]
+@[to_additive, protect_proj]
 structure group_seminorm (G : Type*) [group G] :=
 (to_fun : G → ℝ)
 (map_one' : to_fun 1 = 0)
 (mul_le' : ∀ x y, to_fun (x * y) ≤ to_fun x + to_fun y)
 (inv' : ∀ x, to_fun x⁻¹ = to_fun x)
 
+/-- A nonarchimedean seminorm on an additive group `G` is a function `f : G → ℝ` that preserves
+zero, is nonarchimedean and such that `f (-x) = f x` for all `x`. -/
+@[protect_proj]
+structure nonarch_add_group_seminorm (G : Type*) [add_group G] extends zero_hom G ℝ :=
+(add_le_max' : ∀ r s, to_fun (r + s) ≤ max (to_fun r) (to_fun s))
+(neg' : ∀ r, to_fun (-r) = to_fun r)
+
+/- NOTE: We do not define `nonarch_add_group_seminorm` as an extension of `add_group_seminorm`
+  to avoid having a superfluous `add_le'` field in the resulting structure. The same applies to
+  `nonarch_add_group_norm` below. -/
```

Comment:
I would make that a docstring and possibly move it to the module docs, to increase visibility.
```suggestion
/-! NOTE: We do not define `nonarch_add_group_seminorm` as an extension of `add_group_seminorm`
  to avoid having a superfluous `add_le'` field in the resulting structure. The same applies to
  `nonarch_add_group_norm` below. -/
```